### PR TITLE
📈 Insights QoL Part 2

### PR DESCRIPTION
**Pull request by connorwinston**
- Changed the Panel Selection UI to match the field Selection UI...aka preview.SVG's added.
- Fixed Table Sorting of Dashboards
- Added Bulk Actions to Dashboards
  - Export
  - Delete
- Added Duplicate in Place in Dashboard More Actions Row Dropdown
- Added Export in Dashboard More Actions Row Dropdown
- Added Import to Sidebar for importing exported dashboards

### CODE REVIEW

File: app/src/modules/insights/routes/overview.vue [hunk 1]
```diff
@@ -187,6 +419,23 @@ async function deleteDashboard() {
 </template>
 
 <style scoped>
+.v-input.search {
+	height: var(--v-button-height);
+	--border-radius: calc(44px / 2);
+	width: 200px;
+	margin-left: auto;
+
+	@media (min-width: 600px) {
```
Comment (rijkvanzanten) [inline app/src/modules/insights/routes/overview.vue:428]: Style should be updated to SCSS to use nested values

File: app/src/modules/insights/routes/overview.vue [hunk 2]
```diff
@@ -116,51 +295,104 @@ async function deleteDashboard() {
...
+									<v-icon name="edit" />
+								</v-list-item-icon>
+								<v-list-item-content>
+									{{ t('edit_dashboard') }}
+								</v-list-item-content>
+							</v-list-item>
+
+							<v-list-item class="warning" clickable @click="duplicateDashboard(item.id, toggle)">
+								<v-list-item-icon>
+									<v-progress-circular v-if="duplicating == item.id" indeterminate small />
```
Comment (rijkvanzanten) [inline app/src/modules/insights/routes/overview.vue:345]: prefer `===` over `==`

File: app/src/styles/_colors.scss [hunk 3]
```diff
@@ -29,16 +30,23 @@ $color-mapping: (
...
 	'10': 10,
+	'20': 20,
 	'25': 25,
+	'30': 30,
+	'40': 40,
 	'50': 50,
+	'60': 60,
+	'70': 70,
 	'75': 75,
+	'80': 80,
```
Comment (rijkvanzanten) [inline app/src/styles/_colors.scss:49]: This isn't really going to persist long enough.. These color palettes are scheduled to disappear, so lets try to find an alternative solution

File: app/src/modules/insights/routes/panel-configuration.vue [hunk 4]
```diff
@@ -112,103 +141,257 @@ const stageChanges = () => {
...
+}
+
+.interface.gray .preview {
+	--primary: var(--gray);
+	--primary-20: var(--gray-20);
+	--primary-30: var(--gray-30);
+	--primary-40: var(--gray-40);
+	--primary-50: var(--gray-50);
+	--primary-70: var(--gray-70);
+	--primary-80: var(--gray-80);
```
Comment (rijkvanzanten) [inline app/src/modules/insights/routes/panel-configuration.vue:352]: Lets update the SVGs to use the theme--primary colors instead to make sure this remains long-term compatible

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
Couple of small tweaks. Biggest work will be to simplify the svgs to only use 4 shades of primary, and update the variables to match

File: app/src/panels/meter/preview.svg [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        <path
+            d="M118 64C118 58.7471 116.965 53.5457 114.955 48.6927C112.945 43.8396 109.999 39.4301 106.284 35.7157C102.57 32.0014 98.1604 29.055 93.3073 27.0448C88.4543 25.0346 83.2529 24 78 24C72.7471 24 67.5457 25.0346 62.6927 27.0448C57.8396 29.055 53.4301 32.0014 49.7157 35.7157C46.0014 39.4301 43.055 43.8396 41.0448 48.6927C39.0346 53.5457 38 58.7471 38 64L54 64C54 60.8483 54.6208 57.7274 55.8269 54.8156C57.033 51.9038 58.8008 49.258 61.0294 47.0294C63.258 44.8008 65.9038 43.033 68.8156 41.8269C71.7274 40.6208 74.8483 40 78 40C81.1517 40 84.2726 40.6208 87.1844 41.8269C90.0962 43.033 92.742 44.8008 94.9706 47.0294C97.1992 49.258 98.967 51.9038 100.173 54.8156C101.379 57.7274 102 60.8483 102 64H118Z"
+            fill="var(--theme--primary-subdued)" />
+        <path
+            d="M56.2145 30.4532C50.5639 34.1227 45.9317 39.1601 42.7479 45.098C39.564 51.0358 37.9316 57.6817 38.0022 64.4189L54.0013 64.2513C53.959 60.209 54.9384 56.2215 56.8487 52.6588C58.759 49.0961 61.5383 46.0736 64.9287 43.8719L56.2145 30.4532Z"
+            fill="color-mix(in srgb, var(--theme--background), var(--theme--primary) 25%)" />
```
Comment (ConnorSimply) [inline app/src/panels/meter/preview.svg:15]: @rijkvanzanten Your new theming doesn't have a 25% equivalent of primary, should I include the custom color mixin here or ...thoughts?

File: app/src/panels/pie-chart/preview.svg [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            <rect x="18" y="10" width="120" height="76" rx="6" fill="var(--card-face-color)" class="glow" />
+            <rect x="19" y="11" width="118" height="74" rx="5" stroke="var(--theme--primary)"
+                stroke-width="2" />
+        </g>
+        <rect x="88" y="39" width="24" height="4" rx="1" fill="var(--theme--primary-subdued)" />
+        <rect x="88" y="46" width="41" height="4" rx="1" fill="var(--theme--primary-subdued)" />
+        <rect x="88" y="53" width="26" height="4" rx="1" fill="var(--theme--primary-subdued)" />
+        <path
+            d="M81 48C81 42.8403 79.4035 37.807 76.4296 33.5906C73.4557 29.3742 69.2501 26.1813 64.3896 24.4498C59.5292 22.7182 54.252 22.533 49.2821 23.9195C44.3122 25.306 39.8931 28.1962 36.6309 32.1938L56 48H81Z"
+            fill="color-mix(in srgb, var(--theme--background), var(--theme--primary) 75%)" />
```
Comment (ConnorSimply) [inline app/src/panels/pie-chart/preview.svg:13]: @rijkvanzanten Same thing as other, there is no 75% of primary in your theming, thoughts?

File: app/src/panels/meter/preview.svg (hunk 5 above)
Comment (rijkvanzanten) [reply app/src/panels/meter/preview.svg:15]: This custom mixin works for me!

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**